rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -101,7 +111,36 @@ public BookKeeper(Configuration conf, MetricRegistry metrics) throws FileNotFoun
    */
   private void initializeMetrics()
   {
+    cacheEvictionCount = metrics.counter(METRIC_BOOKKEEPER_CACHE_EVICTION_COUNT);
     localCacheCount = metrics.counter(METRIC_BOOKKEEPER_LOCAL_CACHE_COUNT);
+    localRequestCount = metrics.counter(METRIC_BOOKKEEPER_LOCAL_REQUEST_COUNT);
+    nonlocalRequestCount = metrics.counter(METRIC_BOOKKEEPER_NONLOCAL_REQUEST_COUNT);
+    remoteRequestCount = metrics.counter(METRIC_BOOKKEEPER_REMOTE_REQUEST_COUNT);
+
+    metrics.register(METRIC_BOOKKEEPER_CACHE_HIT_RATE_GAUGE, new Gauge<Double>()
+    {
+      @Override
+      public Double getValue()
+      {
+        return ((double) localRequestCount.getCount() / (localRequestCount.getCount() + remoteRequestCount.getCount()));
+      }
+    });
+    metrics.register(METRIC_BOOKKEEPER_CACHE_MISS_RATE_GAUGE, new Gauge<Double>()
+    {
+      @Override
+      public Double getValue()
+      {
+        return ((double) remoteRequestCount.getCount() / (localRequestCount.getCount() + remoteRequestCount.getCount()));
+      }
+    });
+    metrics.register(METRIC_BOOKKEEPER_CACHE_SIZE_GAUGE, new Gauge<Long>()
+    {
+      @Override
+      public Long getValue()
```

Comment:
This is wrong. `fileMetadataCache.size()` is going to give you the number of files in the cache not the actual size of the cache

Reply:
@abhishekdas99 Changed this (and the corresponding test) to check the actual size of the cache.